src/free.c

```diff
@@ -0,0 +1,56 @@
+#include "malloc.h"
```

Comment:
I would merge this file into malloc.c. They are one thing—allocator. When you change `malloc`, you (almost always) have to change `free`.